tests/draft2020-12/not.json

```diff
@@ -123,5 +123,31 @@
                 "valid": true
             }
         ]
-    }
+    },
+    {
+        "description": "collect annotations inside a 'not', even if collection is disabled",
+        "schema": {
+            "$schema": "https://json-schema.org/draft/2020-12/schema",
+            "not": {
+                "$comment": "this subschema must still produce annotations internally, even though the 'not' will ultimately discard them",
+                "anyOf": [
+                    true,
+                    { "properties": { "foo": true } }
+                ],
+                "unevaluatedProperties": false
+            }
+        },
+        "tests": [
+            {
+                "description": "no properties to evaluate",
+                "data": {},
+                "valid": true
```

Comment:
This should be `false`, right? The `anyOf` passes and the there are no properties so `unevaluatedProperties` passes. Then `not` should fail because it's schema is true.

Reply:
🤔 yeah, I think a passing instance requires a non-`foo` property, e.g. `{ "bar": 1 }`.

Reply:
If possible can we a bit more aggressive at reverting (if not fixing) in these cases? Looks like this stayed in in a broken state until it was rediscovered by someone updating.

Reply:
I was waiting for Karen to respond in case she had some reason why she thought the test should be correct, but she never responded and I forgot about it. Honestly, I think we need to be more careful about merging new tests. Ideally, it shouldn't be merged until at least two implementations run and pass the test.